File: apps/android/app/src/main/java/tech/stygian/presently/data/LocalStoryDraft.kt
package tech.stygian.presently.data

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.UUID

@Entity(tableName = "local_story_drafts")
data class LocalStoryDraft(
    @PrimaryKey val id: String = UUID.randomUUID().toString(),
    val imageData: ByteArray,
    val createdAtEpochMillis: Long = System.currentTimeMillis(),
    val state: String = State.Pending.storageValue,
    val lastError: String? = null,
    val publishedUri: String? = null,
    val publishedCid: String? = null,
    val recordKey: String? = null,
) {
    enum class State(val storageValue: String) {
        Pending("pending"),
        Publishing("publishing"),
        Failed("failed"),
        Published("published"),
    }
}

File: apps/ios/Presently/Models/LocalStoryDraft.swift
import Foundation
import SwiftData

@Model
final class LocalStoryDraft {
    enum State: String, Codable {
        case pending
        case publishing
        case failed
        case published
    }

    @Attribute(.unique) var id: UUID
    @Attribute(.externalStorage) var imageData: Data
    var createdAt: Date
    var stateRawValue: String
    var lastError: String?
    var publishedURI: String?
    var publishedCID: String?
    var recordKey: String?

    var state: State {
        get { State(rawValue: stateRawValue) ?? .pending }
        set { stateRawValue = newValue.rawValue }
    }

    init(
        id: UUID = UUID(),
        imageData: Data,
        createdAt: Date = Date(),
        state: State = .pending,
        recordKey: String? = nil
    ) {
        self.id = id
        self.imageData = imageData
        self.createdAt = createdAt
        stateRawValue = state.rawValue
        self.recordKey = recordKey
    }
}
